docs/decisions/0029-self-service-purchasing-BFF.rst

```diff
@@ -0,0 +1,496 @@
+                    "currency": "usd",
+                    "unit_amount": 1000,
+                    "unit_amount_decimal": "1000"
+                }
+            ]
+        },
+        "field_constraints": {
+          "quantity": { "min": 5, "max": 30 },
+          "enterprise_slug": { "min_length": 3, "max_length": 30, "pattern": "^[a-z0-9-]+$" }
+        }
+    }
+
+As an implementation note for the ``prices`` list, pre-filter the
+Stripe API response to just the Price objects which are supported by the
+checkout flow. I.e. filter prices on the following fields:
+
+- ``active`` = ``true`` (exclude outdated prices)
+
+- ``billing_scheme`` = "per_unit" (exclude Tiered schemes)
+
+- ``livemode`` = ``true`` (exclude test-only prices)
+
+- ``type`` = "recurring" (exclude non-recurring prices)
+
+- ``recurring.usage_type`` =  "licensed" (exclude metered pricing)
+
+The ``field_constraints`` section is likely to contain very slow-moving data,
+but the benefit over static config is that the constraints can be derived
+directly from the backend field definitions which serve as a source of truth.
+This avoids needing to sync constraints across multiple repositories.
+
+**2. Cross-Page Validation Endpoint**
+
+.. code-block::
+
+    POST /api/v1/bffs/checkout/validation
+
+    Authentication: JWT
+    Authorization: Authenticated OR Unauthenticated.
+    Purpose: Validate form fields across all form pages, leveraging LMS API calls to check user existence and slug conflicts
+    Optional Fields: full_name, work_email, company_name, enterprise_slug, stripe_price_id, quantity
+    Side-Effects: None
+    Frontend consumers: Build Trial page, Create Account page
+
+    Request:
+    {
+        "full_name": "John Doe",
+        "work_email": "admin@example.com",
+        "company_name": "Example Corporation",
+        "enterprise_slug": "example-corp",
+        "quantity": 10,
+        "stripe_price_id": "price_1MoBy5LkdIwHu7ixZhnattbh"
+    }
+
+    Response (200 OK - Valid):
+    {
+        "validation_decisions": {
+            "full_name": null,
+            "work_email": null,
+            "company_name": null,
+            "enterprise_slug": null,
+            "quantity": null,
+            "stripe_price_id": null
+        },
+        "user_authn": {
+            "user_exists_for_email": true
+        },
+    }
+
+    Response (400 Bad Request - Validation Errors):
+    {
+        "validation_decisions": {
+            "work_email": {
+                "error_code": "invalid_format",
+                "developer_message": "Email format validation failed"
+            },
+            "enterprise_slug": {
+                "error_code": "existing_enterprise_customer",
+                "developer_message": "The slug conflicts with an existing customer."
+            },
+            "quantity": {
+                "error_code": "range_exceeded",
+                "developer_message": "Quantity 50 exceeds allowed range [5, 30] for stripe_price_id"
```

Comment:
[nit] Idk if there is a way to break this up into 2 seperate error codes (min/max) for instance that way the client can easily parse which error to display back to the user without further parsing.

Reply:
I'd advocate we keep it simple because the localized error message for the user can just say "must be within x and y".